enhancements/cloud-integration/aws/aws-pod-identity.md

```diff
@@ -0,0 +1,569 @@
+would be better (i.e. work in all situations) to create and reconcile the OIDC
+discovery endpoint ourselves as a public S3 bucket.
+
+## Summary
+
+When deployed on AWS and configured to use [bound service
+tokens](https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/bound-sa-tokens.md),
+Kubernetes pods can assume Roles managed by IAM via a binding created by a
+ServiceAccount annotation.  AWS has a
+[webhook](https://github.com/aws/amazon-eks-pod-identity-webhook/) they maintain
+for EKS that does enables this ability.  This enhancement integrates the same
+functionality into OpenShift.
+
+The `cloud-credential-operator` will be responsible for managing this new
+ability as it serves as an alternative credential _injection_ mechanism versus
+the existing mechanism of injecting an IAM user access ID and key into a
+secret.  The existing role/user _creation_ via a CredentialRequest CR is common
+between the methods.
+
+The new injection mechanism works by annotating a `ServiceAccount` with a IAM
+role ARN the pod should assume. When a pod runs with the annotated
+`ServiceAccount`, the webhook injects a environment variables and a projected
+volume into the pod that contains the bound token and Role ARN that can be used
+in AWS SDK client creation.  The AWS SDK searches for this token automatically.
+
+This will be implemented in two phases.
+
+#### Phase 1
+
+Phase 1 will achieve parity with the present-day EKS experience.  This includes
+automatic configuration and deployment of the `aws-pod-identity-webhook` and the
+creation of an OIDC discovery endpoint that AWS STS can use to validate cluster
+created bound service account tokens.  Role creation in IAM and `ServiceAccount`
+annotation will be left as an exercise to the user (as it is in EKS)
+
+#### Phase 2
+
+Phase 2 will make the Role creation and `ServiceAccount` annotation
+automatic, just as the User creation and `Secret` injection are today.  This will
+involve changing the API in the CredentialRequest CRD to allow references to
+`ServiceAccounts` in addition to `Secerts` as an injection mechanism for the
+requested credential.
+
+## Motivation
+
+This allows users to more cleanly express and manage pod identities in IAM when
+running on AWS.  The use of bound tokens to assume roles (vs user id/key) is
+preferred as these tokens can expire and be rotated.
+
+### Goals
+
+#### Phase 1
+
+The OOTB experience with respect to IAM-based pod identity running OpenShift
+on AWS is the same vs EKS.
+
+1. The `aws-pod-identity-webook` is deployed and managed as part of the
+   cluster infrastructure
+1. The cluster creates and managed an OIDC discovery endpoint suitable for STS
+  to do bounded service account token authentication against
+
+#### Phase 2
+
+1. An admin can create a IAM Role via `CredentialRequest` CR and inject (via
+   annotation) to a `ServiceAccount`, just as they can create an IAM user and
+   inject to a `Secret` currently.
```

Comment:
@sjenning we've talked about only having done phase 1 but this Phase 2 is actually complete isn't it?

There's really a phase 3 of transitioning a full OCP install and all components to use this functionality, is that right?